viewing_party/party.py

```diff
@@ -0,0 +1,114 @@
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data,title):
+    wl = user_data["watchlist"]
+    for item in wl:
+        if item["title"] == title:
+            add_to_watched(user_data,wl.pop(wl.index(item)))
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    cumulative_ratings = 0
+    watched = user_data["watched"]
+    for item in watched:
+        cumulative_ratings += item["rating"]
+
+    return cumulative_ratings/len(watched) if len(watched) > 0 else 0
+
+def get_most_watched_genre(user_data):
+    counts = {}
+    watched = user_data["watched"]
+    popular = ['',0]
+    for item in watched:
+        this_genre_count = item["genre"]
+        if this_genre_count not in counts:
+            counts[this_genre_count] = 1
+        else:
+            counts[this_genre_count] += 1
+        if counts[this_genre_count] > popular[1]:
+            popular[0] = this_genre_count
+            popular[1] = counts[this_genre_count]
+
+    return popular[0] if popular[1] > 0 else None
+
+def get_unique_watched(user_data):
+    watched = user_data["watched"]
+    uniques = watched.copy()
+    for friend in user_data["friends"]:
+        for item in friend["watched"]:
+            if item in watched and item in uniques:
+                uniques.remove(item)
+
+    return uniques
+
+def get_friends_unique_watched(user_data):
+    fr_uniques = []
+    watched = user_data["watched"]
+    for friend in user_data["friends"]:
+        for item in friend["watched"]:
+            if item not in watched and item not in fr_uniques:
+                fr_uniques.append(item)
+
+    return fr_uniques
+
+def get_available_recs(user_data):
+    user_watched = user_data["watched"]
+    subscriptions = user_data["subscriptions"]
+    recs = []
+    for friend in user_data["friends"]:
+        for item in friend["watched"]:
+            if item not in user_watched and item["host"] in subscriptions and item not in recs:
+                recs.append(item)
+
+    return recs
+
+def get_new_rec_by_genre(user_data):
+    user_watched = user_data["watched"]
+    num_watched = len(user_watched)
+    recs = []
+    if num_watched > 0:
+        genres = {}
+        top_genre = ""
+        curr_top = ['',0]
+        for i in range(num_watched):
+            current = user_watched[i]["genre"]
+            if current not in genres:
+                genres[current] = 1
+            else:
+                genres[current] += 1
+                if genres[current] > curr_top[1]:
+                    top_genre = current
+                    curr_top[0] = current
+                    curr_top[1] = genres[current]
+
+        for friend in user_data["friends"]:
+            for item in friend["watched"]:
+                if (item["title"] not in user_watched) and item["genre"] == top_genre:
```

Comment:
Could we reuse another of our functions to start us off with a list of movies that our friends watched, but we haven't yet watched?